backend/controllers/program.js

```diff
@@ -1,14 +1,44 @@
 const Program = require('mongoose').model('Program');
+const FocusItem = require('mongoose').model('Focus_Item');
 
 module.exports.getAll = (req, res) => {
 	Program.find({})
-		.then(programs => {
-			return res.status(200).json({
-				status: 'ok',
-				programs: programs ? programs : [], // Ensure we always at the bare minimum send back an empty array
-			});
+		.then(async programs => {
+			// Short circuit the code below
+			if (!programs) {
+				return res.status(200).json({
+					status: 'ok',
+					programs: [],
+				});
+			}
+
+			// Adds a ?focus_items=true query parameter to automatically return all associated focus_items too in a single API call
+			if (req.query.focus_items === 'true') {
+				const populatedPrograms = await Promise.all(
+					programs.map(async p => {
```

Comment:
This allows the Admin UI to access the related `focus_items` for a program without being required to make an additional API call on each page load.